packages/auth-providers/dbAuth/api/src/shared.ts

```diff
@@ -295,12 +295,21 @@ export function getDbAuthResponseBuilder(
     const setCookieHeaders = response.headers?.getSetCookie() || []
 
     if (setCookieHeaders.length > 0) {
-      if ('multiValueHeaders' in event) {
+      delete headers['set-cookie']
+      delete headers['Set-Cookie']
+
+      // `'multiValueHeaders' in event` is true for both Netlify and Vercel
+      // but only Netlify actually supports it. Vercel will just ignore it
+      // https://github.com/vercel/vercel/issues/7820
+      if ('multiValueHeaders' in event && !('x-vercel-id' in headers)) {
```

Comment:
Should we have a helper “supportsMultiVakueHeaders” function and put logic there to test? If need elsewhere or if the Vercel detection changes?

Reply:
Great idea. Done ✅